src/wildcard-service/init.lua

```diff
@@ -1,9 +1,182 @@
+local http_ng = require('resty.http_ng')
+local cache_store = require('resty.http_ng.cache_store')
+local cache_backend = require('resty.http_ng.backend.cache')
+local lrucache = require('resty.lrucache')
+local resty_env = require('resty.env')
+local ngx_re = require('ngx.re')
+local cjson = require('cjson')
+local resty_url = require('resty.url')
+local format = string.format
+
 local _M = {
-  _VERSION = '0.1'
+  _VERSION = '0.1',
+  http_cache = cache_store.new(),
+  sso_cache = lrucache.new(100),
+  domain_cache = lrucache.new(100),
 }
 
 local mt = {
   __index = _M
 }
 
+function _M.new(options)
+  local opts = options or {}
+  local backend = cache_backend.new(opts.client or resty_backend, { cache_store = _M.http_cache })
+
+  local http_client = http_ng.new({
+    backend = backend,
+    options = { ssl = { verify = resty_env.enabled('OPENSSL_VERIFY') or false } }
+  })
+
+  local api_host = opts.api_host or getenv('API_HOST') or 'https://multitenant-admin.3scale.net'
+  local access_token = opts.access_token or getenv('MASTER_ACCESS_TOKEN')
+
+  return setmetatable({
+    options = opts,
+    http_client = http_client,
+    api_host = api_host,
+    access_token = access_token
+  }, mt)
+end
+
+local function get_sso_token(service)
```

Comment:
This does not have to do SSO at all. Access token is generated on the installation, so router can have access to MT API with this token.